### PR TITLE
Upgrade/upgrade to java 17

### LINKED ISSUE #49: Upgrade to jopa 1.1.4

### PR DESCRIPTION
**Pull request by kostobog**
Fix #49
Upgrade includes:
- java 17
  - refactor javax packages to jakatra packages 
- upgrade jopa to 1.1.4
- springboot 3.1.5
  - refactor deprecated security 
- lombok 1.18.30
-

### CODE REVIEW

File: src/main/java/cz/cvut/kbss/analysis/persistence/MainPersistenceFactory.java [hunk 1]
```diff
@@ -66,7 +67,7 @@ private void init() {
         properties.put(ONTOLOGY_PHYSICAL_URI_KEY, repositoryConf.getUrl());
         properties.put(DATA_SOURCE_CLASS, persistenceConf.getDriver());
         properties.put(LANG, persistenceConf.getLanguage());
-        properties.put(CACHE_ENABLED, "false");
+//        properties.put(CACHE_ENABLED, Boolean.FALSE.toString());
```
[the remark this answers is not in the document]
Comment (kostobog) [reply src/main/java/cz/cvut/kbss/analysis/persistence/MainPersistenceFactory.java:70]: @blcham 
I commented the line mistake. 
Should fix it by uncommenting or should I uncomment and set `CACHE_ENABLED` to `true?`

Originally `CACHE_ENABLED` was `false`

File: src/main/java/cz/cvut/kbss/analysis/config/SecurityConfig.java [hunk 2]
```diff
@@ -49,20 +55,22 @@ CorsConfigurationSource corsConfigurationSource() {
...
-                .and()
+//                .requestMatcher(r -> false)
+                .cors(auth -> auth.configurationSource(corsConfigurationSource()))
+                .httpBasic(basic -> basic.disable())
+                .csrf(csrf -> csrf.disable())
+                .sessionManagement(sessionManagement -> sessionManagement.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers(AntPathRequestMatcher.antMatcher("/auth/register")).permitAll()
+                        .requestMatchers(AntPathRequestMatcher.antMatcher("/auth/signin")).permitAll()
+                        .anyRequest().authenticated())
```
Comment (chekaval) [inline src/main/java/cz/cvut/kbss/analysis/config/SecurityConfig.java:72]: I think there should be something like `authenticationManager(buildAuthenticationManager(http))`. But, my changes in the [PR](https://github.com/kbss-cvut/fta-fmea/pull/50) affect this anyway, so it doesn't really matter.

**Comment by chekaval** [comment]
There are no further suggestions from my side neither. I am not sure how the repository is set up (whether my approval will lead to auto-merge into master branch), so I leave it as it is.

File: src/main/java/cz/cvut/kbss/analysis/persistence/MainPersistenceFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (blcham) [reply src/main/java/cz/cvut/kbss/analysis/persistence/MainPersistenceFactory.java]: done

File: src/main/java/cz/cvut/kbss/analysis/config/SecurityConfig.java (hunk 2 above)
Comment (kostobog) [reply src/main/java/cz/cvut/kbss/analysis/config/SecurityConfig.java:71]: For it works without calling`authenticationManager(buildAuthenticationManager(http))` on the filter chain. It seams that the authentication provider bean is automatically set up in the filter chain. 

Also, I removed the line with the unused `authManager` variable :
`final AuthenticationManager authManager = buildAuthenticationManager(http);`

File: src/main/java/cz/cvut/kbss/analysis/persistence/MainPersistenceFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kostobog) [reply src/main/java/cz/cvut/kbss/analysis/persistence/MainPersistenceFactory.java]: Uncommented, `CACHE_ENABLED=false`

**Comment by kostobog** [comment]
Commit [eb91ec1](https://github.com/kbss-cvut/fta-fmea/pull/48/commits/eb91ec1cf6c737d639651886c2143c8bc731f1fc) fixes [Use Springboot in the Dockerfile #30](../issues/30)

**Pull request merged**